presto-main/src/main/java/io/prestosql/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1274,6 +1278,13 @@ protected Scope visitJoin(Join node, Optional<Scope> scope)
             Scope left = process(node.getLeft(), scope);
             Scope right = process(node.getRight(), isLateralRelation(node.getRight()) ? Optional.of(left) : scope);
 
+            if (isLateralRelation(node.getRight()) && (node.getType().equals(RIGHT) || node.getType().equals(FULL))) {
```

Comment:
Can you see if this handles `FROM a, b, LATERAL (...)` where the subquery references columns from `a`? I *think* it should, since joins are parsed left-associative, but I'd like to be sure.

Reply:
Yes, indeed this is the case for IMPLICIT joins. Since they are parsed rightwards, the LATERAL relation is analysed in the scope established by (a JOIN b).
Here's an example:
```
presto:default> SELECT * FROM (VALUES 1) a(x), (VALUES 2) b(y), LATERAL (SELECT 3 WHERE x=1) c(z);
 x | y | z
---+---+---
 1 | 2 | 3
(1 row)
```

In the similar case, but with explicit CROSS JOINs:
`FROM a CROSS JOIN b CROSS JOIN LATERAL c (...)`
 the parsing order and the result is the same.

Taken that implicit joins and CROSS joins are parsed left-associative, there are no more "hidden" rules to resolving chained JOINs (i.e. the parsing order results from the query itself).
I checked different queries with multiple JOINs against LATERAL relation and they all work as expected.